File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int i = 0;
        int l = nums.size()-1;
        while(i<=l){
            // cout<<"nums[i] ="<<nums[i]<<endl;
            // cout<<"nums[l] ="<<nums[l]<<endl;
            if(nums[i]==target ){
                return i;
            }
            if(nums[l]==target){
                return l;
            }
            i++;
            l--;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        
        for i, num in enumerate(nums):
            x = -num
            y = i+1
            z = len(nums)-1

            while y < z :
                ans = nums[y] + nums[z]
                # print
                print(x, nums[y], nums[z], ans)

                if ans == x:
                    vec = [-x,nums[y],nums[z]]
                    if vec not in res:
                        res.append(vec)
                
                if ans < x:
                    y += 1
                else:
                    z -= 1


        return res